clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -1033,10 +1033,18 @@ public Deque<ProducerBatch> getDeque(TopicPartition tp) {
      * Get the deque for the given topic-partition, creating it if necessary.
      */
     private Deque<ProducerBatch> getOrCreateDeque(TopicPartition tp) {
-        TopicInfo topicInfo = topicInfoMap.computeIfAbsent(tp.topic(), k -> new TopicInfo(logContext, k, batchSize));
+        TopicInfo topicInfo = topicInfoMap.computeIfAbsent(tp.topic(),
+                k -> new TopicInfo(createBuiltInPartitioner(logContext, k, batchSize)));
         return topicInfo.batches.computeIfAbsent(tp.partition(), k -> new ArrayDeque<>());
     }
 
+    /**
+     * Subclass can custom {@link BuiltInPartitioner}
```

Comment:
this comment is a bit redundant, since it just describe what the code. It would be better to explain why we need this method rather than inline it. Or we can remove the docs

Reply:
You are right,it seem to don’t need explaining. I removed docs